worker/baggageclaim/api/volume_server_test.go

```diff
@@ -1130,17 +1131,14 @@ var _ = Describe("Volume Server", func() {
 			}
 		})
 
-		It("returns 404 when source path is invalid", func() {
-			request, _ := http.NewRequest("PUT", fmt.Sprintf("/volumes/%s/stream-p2p-out?path=%s&streamInURL=some-url&encoding=gzip", myVolume.Handle, "bogus-path"), nil)
+		It("returns an error when source path is invalid", func() {
+			streamInP2pURL := fmt.Sprintf("%s/volumes/%s/stream-in?path=dest-path", otherWorker.URL, myVolume.Handle)
+			request, _ := http.NewRequest("PUT", fmt.Sprintf("/volumes/%s/stream-p2p-out?path=%s&streamInURL=%s&encoding=gzip", myVolume.Handle, "bogus-path", streamInP2pURL), nil)
 			request.Header.Set("Accept-Encoding", string(baggageclaim.GzipEncoding))
 			recorder := httptest.NewRecorder()
 			handler.ServeHTTP(recorder, request)
-			Expect(recorder.Code).To(Equal(404))
-
-			var responseError *api.ErrorResponse
-			err := json.NewDecoder(recorder.Body).Decode(&responseError)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(responseError.Message).To(Equal("no such file or directory"))
+			Expect(recorder.Code).To(Equal(200)) // status code should always be 200, error is in body
```

Comment:
Is it possible to maintain the way it returns status code as before like 403 404 and 500?

Reply:
I don't think so. With the httplib, once we write to http response writer, if status is not set, the lib will auto set 200 as status. So, here 200 just means p2p streaming successfully started, actually result will be carried in response body. Actually, I think the new way is better. Because response body may take more descriptive error message than http code does, so that ATC log may provide more accurate error description.